interfaces/udev/backend.go

```diff
@@ -174,25 +174,23 @@ func (b *Backend) Setup(appSet *interfaces.SnapAppSet, opts interfaces.Confineme
 		}
 	}
 
-	var deviceBuf bytes.Buffer
-	deviceBuf.WriteString("# This file is automatically generated.\n")
-
+	var devCgroupOpts cgroup.SnapDeviceCgroupOptions
 	if udevSpec.ControlsDeviceCgroup() {
 		// The spec states that the snap can manage its own device
 		// cgroup (typically applies to container-like snaps), in which
 		// case leave a flag for snap-confine in at a known location.
-		deviceBuf.WriteString("# snap is allowed to manage own device cgroup.\n")
-		deviceBuf.WriteString("self-managed=true\n")
+		devCgroupOpts.SelfManaged = true
 	}
 	if (opts.DevMode || opts.Classic) && !opts.JailMode {
 		// Allow devmode
-		deviceBuf.WriteString("# snap uses non-strict confinement.\n")
-		deviceBuf.WriteString("non-strict=true\n")
+		devCgroupOpts.NonStrict = true
 	}
+	// This method never fails so we just ignore the error.
```

Comment:
why not just check the error given that it is returned anyway even if it never fails internally?

Reply:
might be helpful if some programming error messes something up so it would be caught by unit tests or logged

Reply:
I didn't want to do it because it's a never-tested error path but I think I can just return err and carry on.